packages/react-core/src/demos/examples/AlertGroup/AlertGroupToastWithNotificationDrawer.tsx

```diff
@@ -0,0 +1,365 @@
+    setNotifications(prevNotifications => prevNotifications.filter(notification => notification.key !== key));
+  };
+
+  const removeAllNotifications = () => {
+    setNotifications([]);
+  };
+
+  const isNotificationRead = (key: React.Key) =>
+    notifications.find(notification => notification.key === key)?.isNotificationRead;
+
+  const markNotificationRead = (key: React.Key) => {
+    setNotifications(prevNotifications =>
+      prevNotifications.map(notification =>
+        notification.key === key ? { ...notification, isNotificationRead: true } : notification
+      )
+    );
+  };
+
+  const markAllNotificationsRead = () => {
+    setNotifications(prevNotifications =>
+      prevNotifications.map(notification => ({ ...notification, isNotificationRead: true }))
+    );
+  };
+
+  const getUnreadNotificationsNumber = () =>
+    notifications.filter(notification => notification.isNotificationRead === false).length;
+
+  const onNotificationBadgeClick = () => {
+    removeAllAlerts();
+    setDrawerExpanded(!isDrawerExpanded);
+  };
+
+  const onDropdownToggle = (id: React.Key, isActive: boolean) => {
+    if (isActive) {
+      setOpenDropdownKey(id);
+      return;
+    }
+    setOpenDropdownKey(null);
+  };
+
+  const onDropdownSelect = () => {
+    setOpenDropdownKey(null);
+  };
+
+  const buildOverflowMessage = () => {
+    const overflow = alerts.length - maxDisplayed;
+    if (overflow > 0 && maxDisplayed > 0) {
+      return `View ${overflow} more notification(s) in notification drawer`;
+    }
+    return '';
+  };
+
+  const getUniqueId = () => new Date().getTime();
+
+  const getTimeCreated = () => {
+    const dateCreated = new Date();
+    return (
+      dateCreated.toDateString() +
+      ' at ' +
+      ('00' + dateCreated.getHours().toString()).slice(-2) +
+      ':' +
+      ('00' + dateCreated.getMinutes().toString()).slice(-2)
+    );
+  };
+
+  const removeAlert = (key: React.Key) => {
+    setAlerts(prevAlerts => prevAlerts.filter(alert => alert.props.id !== key.toString()));
+  };
+
+  const removeAllAlerts = () => {
+    setAlerts([]);
+  };
+
+  const onAlertGroupOverflowClick = () => {
+    removeAllAlerts();
+    setDrawerExpanded(true);
+  };
+
+  const onMaxDisplayedAlertsMinus = () => {
+    setMaxDisplayed(normalizeAlertsNumber(maxDisplayed - 1));
+  };
+
+  const onMaxDisplayedAlertsChange = (event: any) => {
+    setMaxDisplayed(normalizeAlertsNumber(Number(event.target.value)));
+  };
+
+  const onMaxDisplayedAlertsPlus = () => {
+    setMaxDisplayed(normalizeAlertsNumber(maxDisplayed + 1));
+  };
+
+  const normalizeAlertsNumber = (value: number) => Math.max(Math.min(value, maxAlerts), minAlerts);
+
+  const alertButtonClasses = 'pf-c-button pf-m-secondary';
+
+  const alertButtonStyle = { marginRight: '8px', marginTop: '8px' };
+
+  const notificationBadge = (
+    <NotificationBadge
+      variant={getUnreadNotificationsNumber() === 0 ? 'read' : 'unread'}
```

Comment:
there is a third `attention` state that we should show if there is a danger alert. I know this is a demo of the alert group, but I think if we're going to show the notification badge based on stuff in the notification drawer, we should make sure this is consistent with our best practices otherwise it could be confusing.

https://github.com/patternfly/patternfly-react/blob/c092c7cfaf877c5783c9ad9db0a8010b8ea023eb/packages/react-core/src/components/NotificationBadge/NotificationBadge.tsx#L11

Reply:
I added also this option, I didn't know we have also `attention` state 😅 Thanks!